### PR TITLE
WSearchRelatedTracksMenu: Elide action text

### PR DESCRIPTION
**Pull request by uklotzde**
TODO:

- [x] Rebase after #3215 has been merged

Turned out to be trickier than expected.

- Separate query field and value by a right arrow an unquote the value
- Elide query values in the middle

Screenshot from 2020-10-29 10-49-40

### CODE REVIEW

**Comment by ronso0** [comment]
works with Qt 5.12.8

it seems to elide in the middle (at least the folder).
I'm not 100% sure how to format that string appropriately, but I think we should prioritize trying to show the entire folder name and elide the path before that. That will _probably_ be redundant with the album name, but I don't see the necessity showig the path: the "Search related.." menu is for triggering searches, not for supplying track info. Users should Properties or Show in File Manager" for that.

**Comment by uklotzde** [comment]
> works with Qt 5.12.8
> 
> it seems to elide in the middle (at least the folder).
> I'm not 100% sure how to format that string appropriately, but I think we should prioritize trying to show the entire folder name and elide the path before that. That will _probably_ be redundant with the album name, but I don't see the necessity showig the path: the "Search related.." menu is for triggering searches, not for supplying track info. Users should Properties or Show in File Manager" for that.

Qt::ElideMiddle has been chosen deliberately, a comment in the code explains why. The text is already split into a non-elided prefix and an elided suffix. Anything else would require a more sophisticated strategy.

But is it worth to complicate the code further? I don't think so. The value should only give a glimpse which query will be executed. No data will be modified. The full text becomes visible in the edit box when triggering the search.

File: src/widget/wsearchrelatedtracksmenu.cpp [hunk 1]
```diff
@@ -1,10 +1,18 @@
...
+#include <QScreen>
+
 #include "track/track.h"
+#include "util/math.h"
 #include "util/qt.h"
+#include "util/widgethelper.h"
 
 namespace {
 
+constexpr double kMaxMenuToAvailableScreenWidthRatio = 0.2; // 20%
```
Comment (Be-ing) [inline src/widget/wsearchrelatedtracksmenu.cpp:12]: I presume this is an arbitrary cutoff that you think looks good? If so, please note in the comment that this is arbitrary.

Comment (uklotzde) [reply src/widget/wsearchrelatedtracksmenu.cpp]: Done.

I also removed the other arbitrary lower bound that is not strictly needed and only complicates the code. Less code is better code.

**Comment by ronso0** [comment]
I just found that `path/to/folder` also finds `path/to/folder copy`.

**Comment by uklotzde** [comment]
> I just found that `path/to/folder` also finds `path/to/folder copy`.

I guess currently we are searching just for the prefix, case-insensitive. This is desired to find files in all subfolders. Maybe we need two different folder searches? Or delimit the folder string with a terminating '/'. I will do that as a first step.

**Comment by uklotzde** [comment]
> > I just found that `path/to/folder` also finds `path/to/folder copy`.
> 
> I guess currently we are searching just for the prefix, case-insensitive. This is desired to find files in all subfolders. Maybe we need two different folder searches? Or delimit the folder string with a terminating '/'. I will do that as a first step.

Done. But we cannot exclude subfolders.

**Comment by poelzi** [comment]
Im a huge fan how the fish shell compresses the folderpath.
It uses only the first letter of each folder. Folders under home start at home with ~/

This is usually enough to know there you are but dont spam the ui with characters.

**Comment by uklotzde** [comment]
This PR is not the place for discussing custom elision algorithms.

**Comment by uklotzde** [comment]
I have already left a TODO comment in the code:
> // TODO: Customize the suffix elision?

**Comment by ronso0** [comment]
Unfortunately this looks quite bold here:
image

I tried the hollow triangle, how does that look for you:
image

**Comment by uklotzde** [comment]
> I tried the hollow triangle, how does that look for you:

Unfortunately, the hollow triangle looks like an ordinary character that doesn't distinguish the different parts *field* vs. *value* of the text very well.

**Comment by ronso0** [comment]
I go along with your screenshot at the top, but it looks different here. I wonder why it looks different eventhough we ship the fonts. maybe it's not included in Ubuntu or Open Sans

**Comment by uklotzde** [comment]
@ronso How can we check that the bundled OpenSans font is actually loaded and used? The log file does not contain any messages.

**Comment by uklotzde** [comment]
Custom fonts are loaded as expected. But there is a system-wide *Open Sans* font in Fedora that might be picked instead.

**Comment by ronso0** [comment]
Naa, it's just that the triangle is not included in Ubuntu or OpenSans, both have a very limited set of glyphs and neither contains the Geometric Shapes range 25A0—25FF.
https://fonts.google.com/specimen/Open+Sans#glyphs
https://fonts.google.com/specimen/Ubuntu#glyphs
I guess the OS goes through the fallback fonts and tries to find one that contains the requested glyph.

So we have to live with that for now.

**Comment by ronso0** [comment]
Does it still look like the screenshot at the top for you?
I tried with Tango (and OS font Open Sans) and it looks very dense and I think this might happen to others, too:
Screenshot_2020-11-02_20-41-34


[With an extra space before the triangle](https://github.com/mixxxdj/mixxx/pull/3213/files#diff-eb2ea44548e9a90738dd43f92cdfb83a4c16539e8f8aac9058513f5a44ac43d8R18) it's much more relaxed:
image

**Comment by Holzhaus** [comment]
I think the triangle looks weird. Why not just a colon (`: `) or a dash (` - `)?

**Comment by ronso0** [comment]
We had the colon earlier, it just doesn't separate the prefix and search string enough, thus the triangle.
Dash is also not adequate IMO. It looks more like `Artist - Title` or a range.
And there's currently no easy way to for example highlight the search term with font styles like 'bold'.
That said: the triangle is an imprvement, unless yo can come up with a better unicode char.

**Comment by Be-ing** [comment]
I think a colon is fine. The triangle is okay, but it's a bit odd because it is similar to the submenu icon.

**Comment by uklotzde** [comment]
To stick with ASCII characters I have replaced the right triangle with the pipe symbol. This is more unlikely to appear in any metadata property than a dash symbol.

**Comment by Be-ing** [comment]
Please post a screenshot when significant GUI changes are made to a PR.

**Comment by uklotzde** [comment]
Screenshot from 2020-11-03 01-25-18

Final version. I will not try any other variants.

**Review by ronso0 — APPROVED** [review]
Thanks, looks good!

**Comment by ronso0** [comment]
tbh I'm not happy with the pipe symbol, but we can tweak this later on.

**Pull request merged**